Feature new property

jListo opened the pull request:
New Property API

Commit:
Initial Commit

Initial commit for adding a new property. NOT FUNCTIONAL

Commit:
Restructure and Bug Fixing

Code was restructured and moved to appropriate sections and it is now
functional.

nirrek commented on index.js:
```diff
@@ -113,6 +113,17 @@ server.route([
     }
   },
   {
+    path: '/users/{userId}/addProperty',
```
We will want to use a better endpoint than this.

It should be a POST request to the `/properties` resource endpoint, as you are adding a property to the properties collection.

Can you have a little read up on RESTful design of web services so you understand why this is the case.

nirrek commented on handlers/propertyAddNew.js:
```diff
@@ -0,0 +1,62 @@
+var database = require('../database.js');
+var pool = database.getConnectionPool();
+var getPhotoUrl = require('../utils.js').getPhotoUrl;
```
Unused import

Commit:
Consolidation and Endpoint

Consolidated new properties being added into properties.js and removed
the old file.

nirrek committed:
(properties) handle invalid owner/tenant emails

nirrek committed:
remove photo url generation

nirrek committed:
Merge branch 'master' into feature-new-property
